src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -1061,19 +1075,19 @@ private static void ReportImplicitImplementationMatchDiagnostics(Symbol interfac
                     if (implicitImplMethod.IsConditional)
                     {
                         // CS0629: Conditional member '{0}' cannot implement interface member '{1}' in type '{2}'
-                        diagnostics.Add(ErrorCode.ERR_InterfaceImplementedByConditional, implicitImpl.Locations[0], implicitImpl, interfaceMethod, implementingType);
+                        diagnostics.Add(ErrorCode.ERR_InterfaceImplementedByConditional, getDiagnosticLocation(implicitImpl), implicitImpl, interfaceMethod, implementingType);
                     }
                     else
                     {
-                        ReportAnyMismatchedConstraints(interfaceMethod, implementingType, implicitImplMethod, diagnostics);
+                        ReportAnyMismatchedConstraints(interfaceMethod, implementingType, implicitImplMethod, getDiagnosticLocation(implicitImpl), diagnostics);
```

Comment:
>getDiagnosticLocation(implicitImpl) [](start = 110, length = 35)

We should avoid eager calculation of location before we determined that there is any diagnostics to report. We probably should make ```getDiagnosticLocation``` a real (non-local) static function and call it inside ReportAnyMismatchedConstraints when location is needed.